test/main.hs

```diff
@@ -92,10 +92,7 @@ byteArrayEqProp = QC.property $ \(xs :: [Word8]) (ys :: [Word8]) ->
   (compareLengthFirst xs ys == EQ) === (byteArrayFromList xs == byteArrayFromList ys)
 
 compareLengthFirst :: [Word8] -> [Word8] -> Ordering
-compareLengthFirst xs ys = case compare (length xs) (length ys) of
-  LT -> LT
-  GT -> GT
-  EQ -> compare xs ys
+compareLengthFirst xs ys = (compare `on` length) <> compare xs ys
 
```

Comment:
That's not going to compile. Missing `xs` and `ys`.

Reply:
Might be able to write `compareLengthFirst = (compare `on` length) <> compare`, but that might go too far.

Reply:
fixed